internal/service/searchdeployment/state_transition_search_deployment.go

```diff
@@ -49,9 +49,9 @@ func WaitSearchNodeDelete(ctx context.Context, projectID, clusterName string, cl
 	return err
 }
 
-func searchDeploymentRefreshFunc(ctx context.Context, projectID, clusterName string, client DeploymentService) retry.StateRefreshFunc {
+func searchDeploymentRefreshFunc(ctx context.Context, projectID, clusterName string, client *admin.APIClient) retry.StateRefreshFunc {
```

Comment:
I think this is fine although an alternative is to pass client.AtlasSearchApi across instead of APIClient, what do you think the pros and cons are for both approaches?

Reply:
good point, giving some thought I would favour passing the more specific interface when possible so it is clear what specific api resource the function uses. Adjusted but let me know if you see a downside to this.

Reply:
yes, i agree, i like the latest code passing AtlasSearchApi